cartographer/mapping/internal/3d/local_trajectory_builder_3d.cc

```diff
@@ -134,111 +133,123 @@ LocalTrajectoryBuilder3D::AddRangeData(
   CHECK(!synchronized_data.ranges.empty());
   CHECK_LE(synchronized_data.ranges.back().point_time.time, 0.f);
   const common::Time time_first_point =
-      current_sensor_time +
+      synchronized_data.time +
       common::FromSeconds(synchronized_data.ranges.front().point_time.time);
   if (time_first_point < extrapolator_->GetLastPoseTime()) {
     LOG(INFO) << "Extrapolator is still initializing.";
     return nullptr;
   }
 
-  std::vector<sensor::TimedPointCloudOriginData::RangeMeasurement> hits =
+  if (num_accumulated_ == 0) {
+    accumulated_point_cloud_origin_data_.clear();
+  }
+
+  synchronized_data.ranges =
       sensor::VoxelFilter(0.5f * options_.voxel_filter_size())
           .Filter(synchronized_data.ranges);
+  accumulated_point_cloud_origin_data_.emplace_back(
+      std::move(synchronized_data));
+  ++num_accumulated_;
 
-  std::vector<transform::Rigid3f> hits_poses;
-  hits_poses.reserve(hits.size());
-  bool warned = false;
-
-  for (const auto& hit : hits) {
-    common::Time time_point =
-        current_sensor_time + common::FromSeconds(hit.point_time.time);
-    if (time_point < extrapolator_->GetLastExtrapolatedTime()) {
-      if (!warned) {
-        LOG(ERROR)
```

Comment:
@wohe shall I keep this warning?

Reply:
Seems like a good idea. Also the handling of time like in #936. Or is there some reason this cannot happen anymore?

Reply:
Ok I implemented something equivalent.